### PR TITLE
json-schema-validator: add JSON_DIAGNOSTICS option

### PR DESCRIPTION
**Pull request by jalapenopuzzle**
Needed for compatibility with the nlohmann_json library option of the same name.

Specify library name and version:  **json-schema-validator/2.2.0**

This pull request addresses issue <https://github.com/conan-io/conan-center-index/issues/18340>.
I am not the author of the json-schema-validator library.

---

- [*] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [*] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [*] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/json-schema-validator/all/conanfile.py [hunk 1]
```diff
@@ -23,10 +23,12 @@ class JsonSchemaValidatorConan(ConanFile):
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "JSON_DIAGNOSTICS": [True, False],
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:26]: I think it's better to use lowercase options like: `json_diagnostics`
```suggestion
        "json_diagnostics": [True, False],
```

File: recipes/json-schema-validator/all/conanfile.py [hunk 2]
```diff
@@ -23,10 +23,12 @@ class JsonSchemaValidatorConan(ConanFile):
...
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "JSON_DIAGNOSTICS": False,
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:31]:
```suggestion
        "json_diagnostics": False,
```

File: recipes/json-schema-validator/all/conanfile.py [hunk 3]
```diff
@@ -83,6 +85,8 @@ def generate(self):
         else:
             tc.variables["JSON_VALIDATOR_BUILD_TESTS"] = False
             tc.variables["JSON_VALIDATOR_BUILD_EXAMPLES"] = False
+        if self.options.JSON_DIAGNOSTICS:
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:88]:
```suggestion
        if self.options.json_diagnostics:
```

File: recipes/json-schema-validator/all/conanfile.py [hunk 4]
```diff
@@ -131,6 +135,8 @@ def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "nlohmann_json_schema_validator")
         self.cpp_info.set_property("cmake_target_name", "nlohmann_json_schema_validator")
         self.cpp_info.libs = ["json-schema-validator" if Version(self.version) < "2.1.0" else "nlohmann_json_schema_validator"]
+        if self.options.JSON_DIAGNOSTICS:
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:138]:
```suggestion
        if self.options.json_diagnostics:
```

**Comment by czoido** [comment]
Hi @jalapenopuzzle,

Thanks a lot for the contribution. I did some suggestion related with the casing, in general the options are lowercase.
Also, I think it would be very nice to use the options_description attribute in this recipe to give a little bit of context of what this option is for. You have one example on how to use it here: https://github.com/conan-io/conan-center-index/blob/194bd247f3f14b597fac63074ba82ad54d20c74b/recipes/continuable/all/conanfile.py#L40-L61

File: recipes/json-schema-validator/all/conanfile.py [hunk 5]
```diff
@@ -23,10 +23,26 @@ class JsonSchemaValidatorConan(ConanFile):
...
+        "json_diagnostics": [True, False],
+    }
+    options_description = {
+        "shared": {
+            "Enables shared library building."
+        },
+        "fPIC": {
+            "Enables position-independent code.",
+            "PIC is a body of machine code that executes properly regardless of its absolute address."
+        },
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:35]: It's fine to omit these two 
```suggestion
```

File: recipes/json-schema-validator/all/conanfile.py [hunk 6]
```diff
@@ -23,10 +23,19 @@ class JsonSchemaValidatorConan(ConanFile):
...
         "shared": [True, False],
         "fPIC": [True, False],
+        "json_diagnostics": [True, False],
+    }
+    options_description = {
+        "json_diagnostics": {
+            "Defines JSON_DIAGNOSTICS=1 for the nlohmann_json library.",
+            "Refer https://json.nlohmann.me/api/macros/json_diagnostics/",
+            "This macro enables extended diagnostics for exception messages."
+        }
```
Comment (czoido) [inline recipes/json-schema-validator/all/conanfile.py:33]:
```suggestion
        "json_diagnostics": (
            "Defines JSON_DIAGNOSTICS=1 for the nlohmann_json library. "
            "Refer https://json.nlohmann.me/api/macros/json_diagnostics/ "
            "This macro enables extended diagnostics for exception messages."
        )
```

**Review by czoido — APPROVED** [review]

**Review by davidsanfal — APPROVED** [review]

**Pull request merged**